internal/services/eventgrid/eventgrid_partner_namespace_resource.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	for _, v := range *ipRules {
+		output = append(output, PartnerInboundIpRuleModel{
+			IpMask: pointer.From(v.IPMask),
+			Action: string(pointer.From(v.Action)),
+		})
+	}
+	return output
+}
+
+func (r EventGridPartnerNamespaceResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerNamespaces
+
+			id, err := partnernamespaces.ParsePartnerNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config EventGridPartnerNamespaceResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `model.Properties` was nil", *id)
+			}
+
+			model := existing.Model
+
+			if metadata.ResourceData.HasChange("local_auth_enabled") {
+				model.Properties.DisableLocalAuth = pointer.To(!config.LocalAuthEnabled)
+			}
+			if metadata.ResourceData.HasChange("inbound_ip_rule") {
+				model.Properties.InboundIPRules = expandPartnerInboundIPRules(config.InboundIPRules)
+			}
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				if config.PublicNetworkAccessEnabled {
+					model.Properties.PublicNetworkAccess = pointer.To(partnernamespaces.PublicNetworkAccessEnabled)
+				} else {
+					model.Properties.PublicNetworkAccess = pointer.To(partnernamespaces.PublicNetworkAccessDisabled)
+				}
+			}
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = pointer.To(config.Tags)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerNamespaceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerNamespaces
+
+			id, err := partnernamespaces.ParsePartnerNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q: %+v", metadata.ResourceData.Id(), err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+			state := EventGridPartnerNamespaceResourceModel{
+				ResourceGroup:        id.ResourceGroupName,
+				PartnerNamespaceName: id.PartnerNamespaceName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.NormalizeNilable(pointer.To(model.Location))
```

Comment:
Since `model.Location` is not a pointer, we would just use `location.Normalize` rather than passing the address

```suggestion
				state.Location = location.Normalize(model.Location)
```